src/Facility.Definition/ServiceFieldValidation.cs

```diff
@@ -0,0 +1,137 @@
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Facility.Definition
+{
+	/// <summary>
+	/// A validation criterion of a DTO field.
+	/// </summary>
+	public sealed class ServiceFieldValidation
+	{
+		/// <summary>
+		/// Creates a validation criterion of a DTO field
+		/// </summary>
+		public ServiceFieldValidation(ServiceAttributeInfo attributeInfo)
+		{
+			IsDefinedEnum = attributeInfo.Parameters.Count == 0;
+
+			foreach (var parameterInfo in attributeInfo.Parameters)
+			{
+				switch (parameterInfo.Name)
+				{
+					case "length":
+						LengthRange = GetRange(attributeInfo, parameterInfo);
+						break;
+					case "count":
+						CountRange = GetRange(attributeInfo, parameterInfo);
+						break;
+					case "value":
+						ValueRange = GetRange(attributeInfo, parameterInfo);
+						break;
+					case "regex":
+						if (s_regex.IsMatch(parameterInfo.Value))
```

Comment:
My suggestion above would drop the slashes, in which case this test won't be needed.